### ISSUE TITLE
Error when importing OAS Spec through Postman API

### ISSUE THREAD

**Issue by bzmw** [a85/POSTMan-Chrome-Extension#7360]
I'm trying to import a file into postman using curl and the public api. The file successfully imports into the postman app when I drag and drop it, but I get an error when using the Public API.

# The command
```
 curl --location --request POST "https://api.getpostman.com/import/openapi" \
 --header "x-api-key: <retracted>" \
 --header "X-Workspace-Id: <retracted>" \
 --form "type=file" \
 --form "input=@dist/Public.yaml" \
 -v
```

# The output
```
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 107.23.78.89...
* TCP_NODELAY set
* Connected to api.getpostman.com (107.23.78.89) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.postman.co
*  start date: Feb 26 00:00:00 2019 GMT
*  expire date: Mar 26 12:00:00 2020 GMT
*  subjectAltName: host "api.getpostman.com" matched cert's "*.getpostman.com"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
> POST /import/openapi HTTP/1.1
> Host: api.getpostman.com
> User-Agent: curl/7.54.0
> Accept: */*
> x-api-key: <retracted>
> X-Workspace-Id: <retracted>
> Content-Length: 527307
> Expect: 100-continue
> Content-Type: multipart/form-data; boundary=------------------------82931f6bbfe94bda
>
< HTTP/1.1 100 Continue
< HTTP/1.1 500 Internal Server Error
< Access-Control-Allow-Origin: *
< Content-Type: application/json; charset=utf-8
< Date: Thu, 19 Sep 2019 19:19:47 GMT
< ETag: W/"52-FlVDCgj4RsC0jGOFSHZ8EoeQdrU"
< Server: nginx
< Vary: Accept-Encoding
< x-frame-options: SAMEORIGIN
< X-RateLimit-Limit: 400
< X-RateLimit-Remaining: 399
< X-RateLimit-Reset: 1568920847
< x-srv-span: v=1;s=f447149edea459e8
< x-srv-trace: v=1;t=84f6e403451e9174
< Content-Length: 82
< Connection: keep-alive
* HTTP error before end of send, stop sending
<
* Closing connection 0
* TLSv1.2 (OUT), TLS alert, Client hello (1):
{"error":{"name":"serverError","message":"Something went wrong with the server."}}
```

**Expected behavior**
I expect the command to return 200 and have the OAS spec uploaded in the workspace specified.

**Additional context**
I have tested the oas spec multiple times in different linters include the postman2openapiv2 repo.
*The oas spec imports successfully into the postman app*

[Public.yaml.zip](https://github.com/postmanlabs/postman-app-support/files/3632845/Public.yaml.zip)

**Comment by phebus** [comment]
I am also seeing this exact issue attempting to import a valid Openapi 3.0 yaml file. If there has been any movement on this outside of this issue please advise. Thanks! I can provide any details necessary if need be.

**Comment by bzmw** [comment]
Nothing yet from the Postman team here or in a support email thread I have :(

**Comment by pdonorio** [comment]
Hello, 
I'm calling today POST on `/import/openapi` with python `requests` library,
passing `{'type': 'json', 'input': OPENAPI_JSON_SPECS_STRING}`

but it always end up with `{'error': {'name': 'serverError', 'message': 'Something went wrong with the server.'}}`

Notes:
- the OPENAPI_JSON_SPECS_STRING I have is generated automatically and is validated on the swagger online tool
- the other calls (GET and POST) on workspaces, collections, etc. do not give any problem.

Could someone help?

**Comment by shreys7** [comment]
Hey @bzmw I tried importing the `Public.yaml` file you provided with the following cURL command
```
curl --location --request POST 'https://api.getpostman.com/import/openapi' \
--header 'x-api-key: <api-key>' \
--header 'X-Workspace-Id: <workspace-id>' \
--form 'type=file' \
--form 'input=@/path/to/Public.yaml'
```
It ran successfully by responding with a 200 code and returning the id of the collection created. Can you try again once to see if it works now?

**Comment by shreys7** [comment]
Hey @phebus Can you provide the file you have been trying to import as well as the exact curl command you ran?

**Comment by shreys7** [comment]
Hey @pdonorio, one thing I noticed is you have provided type as 'json', but you have passed the value as `OPENAPI_JSON_SPECS_STRING` for the input parameter. Can you provide the JSON object itself and not the string to see if it works now?

**Comment by pdonorio** [comment]
> Hey @pdonorio, one thing I noticed is you have provided type as 'json', but you have passed the value as `OPENAPI_JSON_SPECS_STRING` for the input parameter. Can you provide the JSON object itself and not the string to see if it works now?

Hi @shreys7, thanks for replying.
I did a couple of tests today, with the string `OPENAPI_JSON_SPECS_STRING` as it was and with the equivalent JSON with no luck.

Then I did a new debug of the whole workflow step by step 
and found out that for POST calls with python library `requests` 
sometimes need to dump the JSON payload.

This one did the trick and finally, the collection is created from the import.
Totally awesome! :D 🎉
